spec/realtime/auth.test.js

```diff
@@ -517,6 +517,39 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
 		authUrl: echoServer + '/respondwith?status=403'
 	}, true); /* expectFailed: */
 
+	exports.authUrl_403_previously_active = function(test) {
+		if(helper.bestTransport === 'jsonp') {
+			/* auth endpoints don't envelope, so this won't work with jsonp */
+			test.done();
+			return;
+		}
+		var realtime, rest = helper.AblyRest();
+		rest.auth.requestToken(null, null, function(err, tokenDetails) {
+			if(err) {
+				test.ok(false, displayError(err));
+				closeAndFinish(test, realtime);
+				return;
+			}
+
+			var authPath = echoServer + "/?type=json&body=" + encodeURIComponent(JSON.stringify(tokenDetails));
+
+			realtime = helper.AblyRealtime({ authUrl: authPath });
+
+			realtime.connection.on('connected', function() {
+				test.ok(true, 'Connected to Ably using authUrl with TokenDetails JSON payload');
+
+				/* replace the authUrl and reauth */
+				realtime.auth.authorize(null, {authUrl: echoServer + '/respondwith?status=403'}, function(err, tokenDetails) {
+					test.equal(err && err.statusCode, 403, 'Check err statusCode');
+					test.equal(err && err.code, 40300, 'Check err code');
```

Comment:
As per [RSA4d](https://docs.ably.io/client-lib-development-guide/features/#RSA4d), the code should be 80019 right 🤔

Reply:
No. Look closer: in RSA4d, 80019 is specified as the code for the ErrorInfo that is "emitted with the state change and set as the connection errorReason". We check that a couple lines below, and if you add an assertion for the code there you'll find it is indeed 80019. This line is checking a different `err`: the one returned in the callback to `authorize()`.

Conceptually: 80019 is a connection error, meaning ["We can't connect because there was a problem with the client-configured authentication provider"](https://github.com/ably/ably-common/blob/main/protocol/errors.json#L153). That would not be appropriate to return in the callback for authorize(). There you want the original error the authorize actually encountered.

Reply:
So, can we add an assertion for the same there? It's difficult to grasp intent at first glance. Maybe I can quickly push the change.

Reply:
go for it